azkaban-common/src/main/java/azkaban/imagemgmt/models/ImageRampup.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.models;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+
+/**
+ * This class represents rampup information for an image type.
+ */
+public class ImageRampup extends BaseModel {
+
+  // Represents rampup plan id
+  private int planId;
```

Comment:
In case of image management APIs, even though integer ids are used as primary key in all the tables, APIs are mostly build around image type name. Users are less likely to remember or use id as get by id api is given least importance. Instead search API around image type name is provided in most of the cases. Hence, lets not consider changing integer id to string bases id as it will be changes with not much benefit. Moreover, both integer and strings are equally used for ids, and both are having advantages and disadvantages. Integers are easy to remember and string with random key is difficult to remember. One key point to remember about image management API is managing the metadata around images and hence it is more of an Azkaban introduced use case rater than user specific use case.

Reply:
What I'm suggesting is to expose to users existing numeric IDs as strings instead of integers, basically, just a data type modification in the request/responses to users. I'm not suggesting to change the data type in the Java code. This should be fairly easy change to do and it will have a huge positive impact in maintenance in the future.

I would argue integers are more easy to remember than alphanumeric strings because a 5 digits integer is not exactly straightforward to remember. I think overall they both are not human friendly, however there must be a unique way to identify resources in REST APIs and IDs are the way to go sooner or later. Even though image type names are unique we should NOT use them as IDs in the URIs because that leads us to inability of name modification (we're actually suffering from this limitation today with flow names and project names). Uniqueness doesn't mean primary key. There are things we can do to mitigate the readability issues around IDs, for example we can implement alias URIs but there must a canonical uri for every resource using the ID.
   
I agree that IDs are not exactly readable regardless if they are integers (is not easy to remember a 5 digits integer) or alphanumeric strings, however